### PR TITLE
Use SelectionDAG for `-g` builds

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: konan/konan.properties [hunk 1]
```diff
@@ -117,7 +117,7 @@ targetSysRoot.ios_arm64 = target-sysroot-9-ios_arm64
 clangFlags.ios_arm64 = -cc1 -emit-obj -disable-llvm-passes -x ir
 clangNooptFlags.ios_arm64 = -O1
 clangOptFlags.ios_arm64 = -O3
-clangDebugFlags.ios_arm64 = -O0
+clangDebugFlags.ios_arm64 = -O0 -mllvm -fast-isel=false -mllvm -global-isel=false
```
Comment (olonho) [inline konan/konan.properties]: Better somehow mark it with bug reference and REMOVEME.

File: konan/konan.properties [hunk 2]
```diff
@@ -117,7 +117,11 @@ targetSysRoot.ios_arm64 = target-sysroot-9-ios_arm64
 clangFlags.ios_arm64 = -cc1 -emit-obj -disable-llvm-passes -x ir
 clangNooptFlags.ios_arm64 = -O1
 clangOptFlags.ios_arm64 = -O3
-clangDebugFlags.ios_arm64 = -O0
+# We force LLVM to use SelectionDAG because fast isel may miscompile sign extension.
+# Related:
```
Comment (SvyatoslavScherbina) [inline konan/konan.properties]: Isn't original LLVM bug related too?

File: konan/konan.properties [hunk 3]
```diff
@@ -117,7 +117,11 @@ targetSysRoot.ios_arm64 = target-sysroot-9-ios_arm64
...
+# https://youtrack.jetbrains.com/issue/KT-33619
+# https://github.com/JetBrains/kotlin-native/issues/3290
+clangDebugFlags.ios_arm64 = -O0 -mllvm -fast-isel=false -mllvm -global-isel=false
```
Comment (SvyatoslavScherbina) [inline konan/konan.properties]: Why `false` for both of them?

Comment (sbogolepov) [reply konan/konan.properties]: To force LLVM to use the third one -- SelectionDAG.

File: konan/konan.properties (hunk 2 above)
Comment (sbogolepov) [reply konan/konan.properties]: Nope.

File: konan/konan.properties (hunk 3 above)
Comment (olonho) [reply konan/konan.properties]: Looks somewhat fragile.

Comment (sbogolepov) [reply konan/konan.properties]: AFAIK there is no flag to explicitly enable SelectionDAG. 
¯\_(ツ)_/¯

**Pull request merged**